### PR TITLE
Summary updates

### PR DESCRIPTION
**Pull request by jasonmotha**
Results look to be the same when changed to the vectorised version (easier to do than have another for loop or component specification)

Left is vectorised, middle is selecting only geom for the summary and right is old functionality

image

### CODE REVIEW

**Review by joelridden — APPROVED** [review]

File: scripts/summarise_all_rels_im.py [hunk 1]
```diff
@@ -73,60 +78,42 @@
...
 
         rel_im_dfs = []
         for c in im_csv_paths:
-            df = pd.read_csv(c, index_col=0)
-            rel_im_dfs.append(df)
+            df = load_im_file_pd(c)
+            rel_im_dfs.append(df[im_types])
 
-        stations = list(set(rel_im_dfs[0].index))
+        stations = list(set(rel_im_dfs[0].index.get_level_values(0)))
```
Comment (jpa198) [inline scripts/summarise_all_rels_im.py:92]: remove list(set()) add .unique()

File: scripts/summarise_all_rels_im.py [hunk 2]
```diff
@@ -73,60 +78,42 @@
...
         stations.sort()
-
         # check IM types. If unspecified, use all IM_types
         wrong_im_count = 0
```
Comment (sungeunbae) [inline scripts/summarise_all_rels_im.py:95]: Looks like this variable is no longer needed?

**Review by sungeunbae — CHANGES_REQUESTED** [review]

File: scripts/summarise_all_rels_im.py [hunk 3]
```diff
@@ -25,35 +29,36 @@
...
+        const="median",
+        default="mean",
         help="find median (default: mean)",
     )
     parser.add_argument(
         "--im",
         dest="im_types",
         nargs="+",
         type=str,
+        choices=ALL_IMS,
```
Comment (joelridden) [inline scripts/summarise_all_rels_im.py:41]: When using this, I'm having some issues with pSA, as it gets a KeyError with pSA in the columns. Maybe should add each im for pSA with the period to this?

Comment (jpa198) [reply scripts/summarise_all_rels_im.py]: You want to explicitly list all 132 pSA periods?
Can I see the help output for that one? :P

**Review by joelridden — APPROVED** [review]

**Review by jpa198 — APPROVED** [review]

File: scripts/summarise_all_rels_im.py [hunk 4]
```diff
@@ -73,60 +76,46 @@
...
-            modified_im_name = (
-                im_type if im_type[0] != "p" else f"p{im_type[1:].replace('p', '.')}"
-            )
-            df_dict[modified_im_name] = log_mean_im
-            df_dict[f"{modified_im_name}_sigma"] = log_stdev_im
+        merged_im_df = pd.concat(rel_im_dfs, axis=0, keys=range(len(rel_im_dfs)))
+        if stat_fn == "mean":
+            log_mean_im = np.exp(np.log(merged_im_df).mean(level=[1, 2]))
+        else:
+            log_mean_im = np.exp(np.log(merged_im_df).median(level=[1, 2]))
```
[the remark this answers is not in the document]
Comment (sungeunbae) [reply scripts/summarise_all_rels_im.py]: good point

**Review by sungeunbae — APPROVED** [review]

**Review by jpa198 — APPROVED** [review]

**Pull request merged**